hazelcast-jet-core/src/main/java/com/hazelcast/jet/DAG.java

```diff
@@ -229,7 +230,19 @@ public Vertex getVertex(String vertexName) {
 
     @Override
     public String toString() {
-        return "Vertices " + verticesByName + "\nEdges " + edges;
+        final StringBuilder b = new StringBuilder("dag\n");
+        for (Iterator<Vertex> it = iterator(); it.hasNext();) {
+            final Vertex v = it.next();
+            b.append("    .vertex(").append(v.getName());
+            if (v.getLocalParallelism() != -1) {
+                b.append(".localParallelism(").append(v.getLocalParallelism()).append(')');
+            }
```

Comment:
Yes. Less characters don't always mean it's easier to read. Advantage: you clearly know, that it is user-provided text (even though generated in case of j.u.s).